GPU/GLES/DrawEngineGLES.cpp

```diff
@@ -562,6 +562,12 @@ void DrawEngineGLES::DoFlush() {
 		if (vertexCount > 0x10000 / 3)
 			vertexCount = 0x10000 / 3;
 #endif
+		// Update viewport width & height before ST.
```

Comment:
What if we just moved `SoftwareTransform(...)` below ApplyDrawState?

It used to be that way (in fact that's exactly why there's even a ApplyDrawStateLate, which needed to run after), but it was moved in f99fa02ba72146b0803f01371669bdf81aca0bff which might've mainly been about hardware transform.

-[Unknown]

Reply:
Yes,it works,but it's a little complicated in vulkan,because vulkan applies draw state as pipeline key.

Reply:
Hm, right.  Seems like we should ultimately refactor this, probably putting the viewport info on `SoftwareTransformParams` and maybe no longer having an ApplyDrawStateLate...

-[Unknown]